decision-records/0004-data-api.md

```diff
@@ -0,0 +1,113 @@
+
+Note about the date:
+- It is not required that this date match exactly the fetch date of a version. As versions are fetched at periodic interval the version that will be returned will corresond to the version valid for this date range.
+- To have the latest version available, simple use the current date time as parameter.
+
+##### Returns
+
+- A raw markdown content when a version was found for this date
+- A HTTP 404 for a date that is anterior of the first available version
+- A HTTP 400 Bad Request, if the date is in the futur
+
+##### Example
+
+```
+GET /version/Open%20Terms%20Archive/Privacy%20Policy/2023-10-24T14%3A41%3A42.381Z.md
+```
+```markdown
+Privacy Policy
+==============
+
+Last updated: September 26, 2023
+
+This privacy policy explains how, why and under what conditions the Open Terms Archive (OTA) site collects personal information and how it is used. Our privacy policy will change over time. And of course, we also record the changes of [our own documents](https://github.com/OpenTermsArchive/demo-versions/tree/main/Open%20Terms%20Archive) 😉
+
+…
+```
+
+### Solution B:
+
+#### Base URL
+
+`<collection host>/api/:version`
+
+#### Endpoint
+
+##### `GET /version/:serviceId/:termsType/:date`
+
+Return a JSON object containing of the version content valid for the given date along some metadata
+
+##### Parameters
+
+| Parameter | Type   | Description            |
+| --------- | ------ | ---------------------- |
+| serviceId | URL-encoded string | The ID of the service |
+| termsType | URL-encoded string | The name of terms type |
+| date | URL-encoded ISO date string | The date for which you want the version. This date should not be in the futur |
+
+Note about the date:
+- It is not required that this date match exactly the fetch date of a version. As versions are fetched at periodic interval the version that will be returned will corresond to the version valid for this date range.
+- To have the latest version available, simple use the current date time as parameter.
+
+##### Returns
+
+- A raw markdown content when a version was found for this date
+- A HTTP 404 for a date that is anterior of the first available version
+- A HTTP 400 Bad Request, if the date is in the futur
+
+##### Example
+
+```
+GET /version/Open%20Terms%20Archive/Privacy%20Policy/2023-10-24T14%3A41%3A42.381Z
+```
+```json
+{
+  "fetchDate": "2023-10-20T06:30:00.381Z",
+  "id": "c0dac2866fb2cdef7f8b98cc260177ac43df273b",
+  "content": "
+Privacy Policy
+==============
+
+Last updated: September 26, 2023
+
+This privacy policy explains how, why and under what conditions the Open Terms Archive (OTA) site collects personal information and how it is used. Our privacy policy will change over time. And of course, we also record the changes of [our own documents](https://github.com/OpenTermsArchive/demo-versions/tree/main/Open%20Terms%20Archive) 😉
+
+…
+"
```

Comment:
This multi-line example is invalid. We would need to transform newlines to `\n`. The example should also provide an example with JSON data, to demonstrate how it would be encoded.